forwards_paper.tex

```diff
@@ -816,80 +816,134 @@ \subsubsection*{Sequential simplification and prior history}
+Not every new edge derives from a never-before-seen node,
 but the number of nodes is at most equal to the number of edges plus the sample size.
-The space complexity after simplification is therefore $O(N + N\log N)$, since
-we need $O(N)$ space to store the first marginal tree, and $O(1)$ space
-for each of the $O(N \log N)$ trees that follow. Thus, we need $O(N^2)$ space
-to store the complete history of the simulation, but only $O(N + N \log N)$
-space to store the history that is relevant to the extant population.
+Therefore,
+we would need $O(N^2)$ space to store the complete history of the simulation,
+but only $O(N \log N)$ % $O(N + N \log N)$
+to store the history that is relevant to the extant population.
+
 
 %% Progressive memory reduction as a function of T
-\citet{padhukasahasram2008exploring} obtained impressive speedups relative to contemporary simulators
-by retaining only 8 generations of history.
-This begs the question: how fast does the information in the pedigree become irrelevant?
-Suppose instead that we have run a simulation of $2N$ individuals for $T$ generations,
-begun with no prior history.
-One way to answer this question is to ask:
-How many of the resulting $4NT$ edges are required after simplification?
-As above, the expected number of edges is bounded by $2N-2$ plus four times the mean marginal tree length.
-Roughly, the length of time for which a marginal tree has $k$ tips is
-$2N/(k(k-1)) = 2N(1/(k-1) - 1/k)$ generations,
-and so the time to go from $N$ to $n$ tips is $2N(1/(n-1) - 1/(N-1))$.
-This implies that after running for time $T$ we expect to
-have around $n(T)$ roots, where $n(T) \approx 1 + 2N/(T+2)$.
+What if $T$ is smaller:
+how many of the resulting $2NT$ edges are required after simplification?
+In other words, how fast does the information in the pedigree become irrelevant?
+% \citet{padhukasahasram2008exploring} obtained impressive speedups relative to contemporary simulators
+% by retaining only 8 generations of history.
+% This begs the question: how fast does the information in the pedigree become irrelevant?
+Now, we need to compute the expected total length of all branches in a coalescent tree
+up until time $T$ (or the common ancestor, whichever comes first).
+Again, coalescent theory tells us that
+the expected length of time for which a coalescent tree has $k$ lineages
+is $2N/(k(k-1)) = 2N(1/(k-1) - 1/k)$ generations.
+Since the tree has $k$ branches over this period,
+it is expected to contribute $2N/(k-1)$ to the total tree length.
+By summing over $n < k \le N$,
+the $N$ tips of a tree are expected to descend from only $n$ lineages
+around $2(N/n - 1)$ generations ago.
+Inverting this relationship between time and number of roots
+implies that a marginal tree cut $T$ units of time ago
+% since 2N(1/n - 1/N) = 2(N/n-1) = T implies
+%    N/n = (T+2)/2 and so n = 2N/(T+2)
+is expected to have around $r(T)$ roots, where $r(T) \approx 2N/(T+2)$.
```

Comment:
Why the \approx here? We've solved exactly so far, haven't we? (Sure the whole calculation is quite approximate, but this is captured by the 'around' earlier)

Reply:
ah sure, good point.